library/std/src/sys/windows/io.rs

```diff
@@ -78,3 +83,54 @@ impl<'a> IoSliceMut<'a> {
         unsafe { slice::from_raw_parts_mut(self.vec.buf as *mut u8, self.vec.len as usize) }
     }
 }
+
+pub unsafe fn handle_is_console(handle: &BorrowedHandle<'_>) -> io::Result<bool> {
+    let handle = handle.as_raw_handle();
```

Comment:
As stated in the [`BorrowedHandle` docs](https://doc.rust-lang.org/nightly/std/os/windows/io/struct.BorrowedHandle.html), the handle "may have the value `NULL` (0), which can occur when consoles are detached from processes, or when windows_subsystem is used".

Therefore we could immediately return `Ok(false)` in the `NULL` case.

Reply:
Done.